documentation/modules/oauth/con-oauth-authorization-keycloak-authorization-services.adoc

```diff
@@ -0,0 +1,199 @@
+Often, only a subset of these actions is actually permitted.
+
+`Authorization scopes` is simply a set of all available actions on all the different resource types. When defining a new `resource` the `scopes` added there are taken from this set of all scopes.
+
+`Policies` are rules that can use various criteria to match a list of accounts. They can match service accounts based on clientId, they can match user accounts based on username, or group, or roles.
+
+Finally, `Permissions` are used to grant a subset of `authorization scopes` on the specific `resource` definition to a set of users.
+
+== Mapping Keycloak Authorization Services to Kafka Model
+
+In order to grant Kafka permissions to users or service accounts you have to use the Keycloak Authorization Services rules on the OAuth client that represents the Kafka Broker - typically this client has `kafka` as its clientId.
+
+This client needs to have `Authorization Enabled` option activated.
+
+All the permissions exist within the scope of this OAuth client, which means that if you have different Kafka clusters configured with different OAuth client ids they would each have a separate set of permissions even though they are part of the same realm.
+
+When the Kafka client authenticates using `SASL_OAUTHBEARER` the `KeycloakRBACAuthorizer` retrieves the list of grants for the current session from the Keycloak server using the access token of the current session.
+This list of grants is the result of evaluating the Keycloak Authorization Services policies and permissions.
+
+.Authorization scopes
+
+Typically the initial configuration involves uploading the authorization scopes which creates a list of all the possible actions that can be performed on all the Kafka resource types.
+This step is performed only once, before defining any permissions. Alternatively, the authorization scopes can be added manually, but make sure to not introduce typos.
+
+The `authorization scopes` should simply contain all the possible Kafka permissions regardless of the resource type:
+
+* Create
+* Write
+* Read
+* Delete
+* Describe
+* Alter
+* DescribeConfig
+* AlterConfig
+* ClusterAction
+* IdempotentWrite
+
+.Resources
+
+The `resources` use pattern names to be used with pattern matching against the targeted resources when performing the permission check.
+
+The general pattern is as follows: `RESOURCE_TYPE:NAME_PATTERN`
+
+The resource types mirror the Kafka authorization model.
+The pattern allows for the two matching options: exact matching (when the pattern does not end with `\*`), and prefix matching (when the pattern ends with `*`).
+
+A few examples:
+
+    Topic:my-topic
+    Topic:orders-*
+    Group:orders-*
+    Cluster:*
+
+In addition, the general pattern can be prefixed by another one of the format `kafka-cluster:CLUSTER_NAME`, followed by comma, where the cluster name refers to the `metadata.name` in the Kafka CR.
```

Comment:
```suggestion
In addition, the general pattern can be prefixed by another in the format `kafka-cluster:CLUSTER_NAME`, followed by a comma, where the cluster name refers to the `metadata.name` in the Kafka custom resource.
```

Reply:
_In addition_ or _alternatively_?

Reply:
Is there semantic difference or just a matter of how it sounds?